gateway/test/src/test/java/io/apiman/gateway/test/junit/vertx3/Vertx3GatewayFileRegistryServer.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2015 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.apiman.gateway.test.junit.vertx3;
+
+import io.apiman.gateway.engine.vertx.polling.URILoadingRegistry;
+import io.apiman.gateway.platforms.vertx3.verticles.ApiVerticle;
+import io.apiman.gateway.platforms.vertx3.verticles.InitVerticle;
+import io.apiman.test.common.echo.EchoServer;
+import io.apiman.test.common.resttest.IGatewayTestServer;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.eventbus.DeliveryOptions;
+import io.vertx.core.json.JsonObject;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.util.concurrent.CountDownLatch;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+/**
+ * A Vert.x 3 version of the gateway test server
+ *
+ * @author Marc Savy {@literal <msavy@redhat.com>}
+ */
+@SuppressWarnings("nls")
+public class Vertx3GatewayFileRegistryServer implements IGatewayTestServer {
+
+    protected static final int API_PORT = 9009;
```

Comment:
Note the trick here, the port reported to the test framework matches that defined for the emulating registry (`ApiToFileRegistry.java`), however the port given to the registry under test is different - so its methods are never invoked (which makes sense as it's a polling/pull registry).

Somewhat sleight of hand, but works very nicely.